File: 2744-find-maximum-number-of-string-pairs/2744-find-maximum-number-of-string-pairs.js
/**
 * @param {string[]} words
 * @return {number}
 */
var maximumNumberOfStringPairs = function(words) {
  let count = 0;

  for (let i = 0; i < words.length; i++) {
    let word = words[i];
    for (let j = i + 1; j < words.length; j++) {
      let reverseWord = words[j][1] + words[j][0];

      if (word === reverseWord) count++;
    }
  }
  return count;
};

File: 2744-find-maximum-number-of-string-pairs/2744-find-maximum-number-of-string-pairs.py
class Solution:
    def maximumNumberOfStringPairs(self, words: List[str]) -> int:
        count = 0
        for i in range(len(words)):
            for j in range(i + 1, len(words)):
                reverse_word = words[j][::-1]
                if(words[i] == reverse_word):
                    count += 1
        return count